internal/net/tcp/dialer.go

```diff
@@ -81,20 +81,34 @@ func NewDialer(opts ...DialerOption) Dialer {
 		} else {
 			d.dialer = d.der.DialContext
 		}
-		return d
-	}
-	if d.cache == nil {
-		d.cache = gache.New()
+		return d, nil
 	}
 
 	if d.dnsRefreshDuration > d.dnsCacheExpiration {
```

Comment:
![badge](https://img.shields.io/badge/review-ask-blueviolet.svg?style=flat-square)
Can we compare the string values by ">" operater?

Reply:
I'll fix it

Reply:
@rinx fixed